src/Umbraco.Web.UI.Client/src/common/resources/ourpackagerrepository.resource.js

```diff
@@ -36,7 +36,21 @@ function ourPackageRepositoryResource($q, $http, umbDataFormatter, umbRequestHel
                $http.get(baseurl + "?pageIndex=0&pageSize=" + maxResults + "&category=" + category + "&order=Popular&version=" + Umbraco.Sys.ServerVariables.application.version),
                'Failed to query packages');
         },
-       
+
+        getPromoted: function (maxResults, category) {
+
+            if (maxResults === undefined) {
+                maxResults = 20;
+            }
+            if (category === undefined) {
+                category = "";
+            }
```

Comment:
How about this instead:
https://github.com/umbraco/Umbraco-CMS/blob/8eb3ef65f686619d09af478bd6fd563fc90580b2/src/Umbraco.Web.UI.Client/src/common/resources/log.resource.js#L65-L76

```
var defaults = {
       pageSize: 10,
       pageNumber: 1,
};

if (options === undefined) {
   options = {};
}

//overwrite the defaults if there are any specified
Utilities.extend(defaults, options);
//now copy back to the options we will use
options = defaults;
```

or like this which extend the original `config` object:
https://github.com/umbraco/Umbraco-CMS/blob/5bfab13dc5a268714aad2426a2b68ab5561a6407/src/Umbraco.Web.UI.Client/src/views/prevalueeditors/colorpicker.controller.js#L4-L13

Reply:
Might be nicer I agree, but I'm following existing patterns in this file with the changes, so am more comfortable just continuing with what's working already for the other cases.